components/prism-pascal.js

```diff
@@ -5,9 +5,17 @@
 */
 
 Prism.languages.pascal = {
+	'asm': {
+		pattern: /(asm)([\s\S]+?)(end)/i,
+		lookbehind: true,
+		inside:{keyword:/end/i,punctuation:/;$/i}
```

Comment:
1. Formatting.
1. The punctuation rule won't work. Did you mean `/;$/m`?
```suggestion
		inside: {
			'keyword': /end/i,
			'punctuation': /;$/i
		}
```

Reply:
It will be good, but the resulting html code is like this:

```
<span class="token keyword">asm</span>
  <span class="token asm">
    MOV AX,1234h
    MOV Number,AX
  <span class="token keyword">end</span>
  <span class="token punctuation">;</span>
</span>
```

In fact, I think it should be:

```
<span class="token keyword">asm</span>
<span class="token asm">
  MOV AX,1234h
  MOV Number,AX
</span>
<span class="token keyword">end</span>
<span class="token punctuation">;</span>
```

But I do not know how to do it.

Reply:
In that case, we can make the whole `asm` rule like this:

```js
'asm: {
    pattern: /(asm\b)[\s\S]+?(?=\bend;)/,
    lookbehind: true
}
```

Reply:
Works perfect, thank you.